From 12d62804412b7dd7efbf75bc3e712a7ff87c3be8 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Sat, 10 Dec 2016 10:25:46 +0100
Subject: [PATCH 1/2] Allow to define PYTHON_INSTALL

---
 wrappers/pyAgrum/CMakeLists.txt | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/wrappers/pyAgrum/CMakeLists.txt b/wrappers/pyAgrum/CMakeLists.txt
index f3d0174..c00deda 100644
--- a/wrappers/pyAgrum/CMakeLists.txt
+++ b/wrappers/pyAgrum/CMakeLists.txt
@@ -33,11 +33,13 @@ else(FOR_PYTHON2)
 endif(FOR_PYTHON2)
 
 # PYTHON_INSTALL
-execute_process(
-  COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix='${CMAKE_INSTALL_PREFIX}'))"
-  OUTPUT_VARIABLE PYTHON_INSTALL
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+if (NOT DEFINED PYTHON_INSTALL)
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix='${CMAKE_INSTALL_PREFIX}'))"
+    OUTPUT_VARIABLE PYTHON_INSTALL
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+endif ()
 
 set(GENERATED_PYTHON "${CMAKE_CURRENT_SOURCE_DIR}/generated-files${PYTHON_VERSION_MAJOR}")
 set(_PYAGRUMLIB "_pyAgrum")
-- 
2.10.2

