diff --git a/src/agrum/PRM/o3prm/O3prmReader.tcc b/src/agrum/PRM/o3prm/O3prmReader.tcc
index 6741483..3e23f44 100644
--- src/agrum/PRM/o3prm/O3prmReader.tcc
+++ src/agrum/PRM/o3prm/O3prmReader.tcc
@@ -37,9 +37,9 @@ namespace gum {
       O3prmReader<GUM_SCALAR>::__clean( std::string text ) const {
         auto match = std::regex( "Syntax error" );
         if ( std::regex_search( text, match ) ) {
-          text = std::regex_replace( text, match, "Error" );
+          text = std::regex_replace( text, match, std::string("Error") );
           auto regex = std::regex{"[A-Z_][A-Z_]+"};
-          text = std::regex_replace( text, regex, "declaration" );
+          text = std::regex_replace( text, regex, std::string("declaration") );
           return text;
         }
         return text;
